src/package.coffee

```diff
@@ -26,16 +26,12 @@ class Package
     packagePath?.startsWith(@resourcePathWithTrailingSlash)
 
   @normalizeMetadata: (metadata) ->
-    if typeof metadata.repository is 'string'
-      metadata.repository =
-        type: 'git'
-        url: metadata.repository
-
-    repoUrl = metadata.repository?.url
-    if repoUrl
-      info = hostedGitInfo.fromUrl(repoUrl)
-      if info?.getDefaultRepresentation() is 'shortcut'
-        metadata.repository.url = info.https().replace(/^git\+/, '')
+    unless metadata?._id
+      normalizePackageData ?= require 'normalize-package-data'
+      normalizePackageData(metadata)
+      if metadata.repository?.type is 'git'
+        metadata.repository.url = metadata.repository.url.replace(/^git\+/, '')
```

Comment:
I think you may want to check `metadata.repository.url` before calling `replace` on it. I think it could be undefined even with `repository.type` is `'git'`.

```sh
> metadata = {"name": "hey", "version": "1.0.0", "repository": {"type": "git"}}
{ name: 'hey', version: '1.0.0', repository: { type: 'git' } }
> normalize = require('normalize-package-data')
[Function: normalize]
> normalize(metadata)
undefined
> metadata
{ name: 'hey',
  version: '1.0.0',
  repository: { type: 'git' },
  readme: 'ERROR: No README data found!',
  _id: 'hey@1.0.0' }
> metadata.repository.url
undefined
>
```